### ISSUE TITLE
logstash-filter-translate 3.2.0 fails to start

### ISSUE THREAD

**Issue by gyterpena** [elastic/logstash#9936]
Hello

With version 3.2.0 of logstash-filter-translate I get following error when trying to load json dictionary(url.json). This does work on version 3.1.0

- Version: Logstash 6.3.0, logstash-filter-translate 3.2.0
- Operating System: CentOS Linux 7
- Config File
```
        if [http][url_full] and [dst_external] {
            translate {
                id => "enrichment_translate_url"
                field => "[http][url_full]"
                destination => "[lookup][blk_url]"
                dictionary_path => '/etc/logstash/translate/url.json'
                add_tag => [ "blacklist", "url_bla" ]
                refresh_interval => "900"
                refresh_behaviour => replace
            }
        }
```
- Sample Data: url.json(edited with correct sample data)
full dictionary https://github.com/gyterpena/ELK/blob/master/logstash/translate/url.json
```
{
  "http://tvmarket.co.kr/Order/Document.zip?nwKVqqZusTVWWIu_i_d_kauhanen@canotti.biz[Document_Pdf________________________________________________________________.exe]": {
    "dateadded": "2018-07-21 22:45:50",
    "tags": "exe,hancitor",
    "url": "http://tvmarket.co.kr/Order/Document.zip?nwKVqqZusTVWWIu_i_d_kauhanen@canotti.biz[Document_Pdf________________________________________________________________.exe]",
    "urlhaus_link": "https://urlhaus.abuse.ch/url/35015/",
    "threat": "malware_download",
    "urlstatus": "offline",
    "id": "35015"
  }
}
```

- Steps to Reproduce: Try to load url.json dictionary with translate plugin on version 3.2.0

Error:
[2018-08-23T17:14:52,698][ERROR][logstash.pipeline        ] Error registering plugin {:pipeline_id=>"default", :plugin=>"#<LogStash::FilterDelegator:0x1f1a3ab6 @metric_events_out=org.jruby.proxy.org.logstash.instrument.metrics.counter.LongCounter$Proxy2 -  name: out value:0, @metric_events_in=org.jruby.proxy.org.logstash.instrument.metrics.counter.LongCounter$Proxy2 -  name: in value:0, @metric_events_time=org.jruby.pr
oxy.org.logstash.instrument.metrics.counter.LongCounter$Proxy2 -  name: duration_in_millis value:0, @id=\"enrichment_translate_url\", @klass=LogStash::Filters::Translate, @metric_events=#<LogStash::Instrument::N
amespacedMetric:0x425cc125 @metric=#<LogStash::Instrument::Metric:0x40f5097 @collector=#<LogStash::Instrument::Collector:0xdceddcb @agent=nil, @metric_store=#<LogStash::Instrument::MetricStore:0x43cf894e @store=
#<Concurrent::Map:0x00000000000fc4 entries=3 default_proc=nil>, @structured_lookup_mutex=#<Mutex:0x6290e404>, @fast_lookup=#<Concurrent::Map:0x00000000000fc8 entries=511 default_proc=nil>>>>, @namespace_name=[:s
tats, :pipelines, :default, :plugins, :filters, :enrichment_translate_url, :events]>, @filter=<LogStash::Filters::Translate id=>\"enrichment_translate_url\", field=>\"[http][url_full]\", destination=>\"[lookup][
blk_url]\", dictionary_path=>\"/etc/logstash/translate/url.json\", add_tag=>[\"blacklist\", \"url_bla\"], refresh_interval=>900, refresh_behaviour=>\"replace\", enable_metric=>true, periodic_flush=>false, overri
de=>false, exact=>true, regex=>false>>", :error=>"Translate: premature end of char-class: /http:\\/\\/tvmarket.co.kr\\/Order\\/Document.zip?nwKVqqZusTVWWIu_i_d_kauhanen@canotti.biz[Document_Pdf__________________
______________________________________________+.exe%5D/ when loading dictionary file at /etc/logstash/translate/url.json", :thread=>"#<Thread:0x8f6b01e run>"}
[2018-08-23T17:14:52,768][ERROR][logstash.pipeline        ] Pipeline aborted due to error {:pipeline_id=>"default", :exception=>#<LogStash::Filters::Dictionary::DictionaryFileError: Translate: premature end of c
har-class: /http:\/\/tvmarket.co.kr\/Order\/Document.zip?nwKVqqZusTVWWIu_i_d_kauhanen@canotti.biz[Document_Pdf________________________________________________________________+.exe%5D/ when loading dictionary fil
e at /etc/logstash/translate/url.json>, :backtrace=>["/usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-translate-3.2.0/lib/logstash/filters/dictionary/file.rb:134:in `loading_exception'", "/usr
/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-translate-3.2.0/lib/logstash/filters/dictionary/file.rb:69:in `load_dictionary'", "/usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filte
r-translate-3.2.0/lib/logstash/filters/dictionary/file.rb:52:in `initialize'", "/usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-translate-3.2.0/lib/logstash/filters/dictionary/file.rb:17:in `c
reate'", "/usr/share/logstash/vendor/bundle/jruby/2.3.0/gems/logstash-filter-translate-3.2.0/lib/logstash/filters/translate.rb:166:in `register'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:340:in `register_plugin'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:351:in `block in register_plugins'", "org/jruby/RubyArray.java:1734:in `each'", "/usr/share/logstash/logstash-core/lib/logstash/
pipeline.rb:351:in `register_plugins'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:729:in `maybe_setup_out_plugins'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:361:in `start_wo
rkers'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:288:in `run'", "/usr/share/logstash/logstash-core/lib/logstash/pipeline.rb:248:in `block in start'"], :thread=>"#<Thread:0x8f6b01e run>"}

**Comment by guyboertje** [comment]
@gyterpena 
I edited your original message to put the "code" pieces into triple backticks ``` sections.
This shows that there are escaped double quotes in the json source.

But the error message talks of `.exe%5D` and %5D is a closing square bracket - we can't see this character in the sample JSON.

**Comment by gyterpena** [comment]
@guyboertje closing ] and opening ] are in the url.json after @canotti.biz [Document_Pdf________________________________________________________________.exe]

**Comment by guyboertje** [comment]
There is an explanation. `premature end of char-class` is an error from Ruby's Regex class.
There is a bug in 3.2.[0,1,2] that is always reacting as if `exact` and `regex` are both true. This means that Regex instances were being created from your JSON keys - and the keys will not have been escaped properly.
I have fixed this locally. A PR will be up soon.

**Comment by gyterpena** [comment]
Looks like this was fixed in logstash-filter-translate 3.2.2, closing.
